usecase/reservation/reservation.go

```diff
@@ -0,0 +1,36 @@
+package reservation
+
+import (
+	"room_app_back/domain/model"
+	"room_app_back/domain/repository/reservation"
+)
+
+type ReservationUsecase struct {
+	rr reservation.ReservationRepository
+}
+
+func NewReservationUsecase(rr reservation.ReservationRepository) *ReservationUsecase {
+	return &ReservationUsecase{
+		rr: rr,
+	}
+}
+
+func (ru *ReservationUsecase) Add(r model.Reservation) (*model.Reservation, error) {
```

Comment:
今の実装だと、Controller層がDomain層を参照できてしまっているので、UseCaseにI/Oを作ってあげた方が良さそう。

```suggestion
func (ru *ReservationUsecase) Add(input *AddInput) (*AddOutput, error) {
```

Reply:
https://github.com/temmie524/room-management-app-backend/pull/25/commits/3f49e4e3fe38b4b5a8b20366bbe20769320b06a1
こちらもひとまずUserのUpdateのみ実装しました。動作確認もできています。

Reply:
- UserI/O続き。他の関数にも実装。
[cda2139](https://github.com/temmie524/room-management-app-backend/pull/25/commits/cda21396b9a75d81e566d170dff4f5f6d68dc386)
- RoomI/O
 [3571867](https://github.com/temmie524/room-management-app-backend/pull/25/commits/3571867cbebb3161f9ff2fb52a44469550e6ada2)

- ReservationI/O (commitメッセージがRoomになっていますが、正しくはReservationのI/O実装です)
[8f5419f](https://github.com/temmie524/room-management-app-backend/pull/25/commits/8f5419f40925c70aa80b360080c06d774c45ca78)